Casks/c/chessx.rb

```diff
@@ -1,16 +1,24 @@
 cask "chessx" do
-  version "1.6.4"
-  sha256 "cf8d57ceb7ebf400ab16a9b5fe5283dac4b3e80f4702ef5190a81f19c124d57f"
+  version "1.6.4a,1.6.4"
+  sha256 "a452933f0862ecf7df29d3f5018ef0ac554de95eefd3c1114d7984a85c66b324"
 
-  url "https://downloads.sourceforge.net/chessx/chessx/#{version}/chessx-#{version}.dmg",
+  url "https://downloads.sourceforge.net/chessx/chessx/#{version.csv.second || version.csv.first}/chessx-#{version.csv.first}.dmg",
       verified: "downloads.sourceforge.net/chessx/"
   name "ChessX"
   desc "Chess database"
   homepage "https://chessx.sourceforge.io/"
 
   livecheck do
     url "https://sourceforge.net/projects/chessx/rss?path=/chessx"
-    regex(%r{url=.*?/chessx[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
+    regex(%r{url=.*?/v?(\d+(?:\.\d+)+)/chessx[._-]v?(\d+(?:\.\d+)+\w?)\.dmg}i)
+    strategy :page_match do |page, regex|
+      match = page.match(regex)
+      next if match.blank?
+
+      next match[2] if match[1] == match[2]
+
+      "#{match[2]},#{match[1]}"
```

Comment:
```suggestion
    regex(%r{url=.*?/v?(\d+(?:\.\d+)+)/chessx[._-]v?(\d+(?:\.\d+)+[a-z]?)\.dmg}i)
    strategy :page_match do |page, regex|
      match = page.match(regex)
      next if match.blank?

      (match[1] == match[2]) ? match[2] : "#{match[2]},#{match[1]}"
```

While `\w` is convenient, it's equivalent to `[a-z0-9_-]` [in a case-insensitive regex], so there's something to be said about using `[a-z]` for the sake of being explicit and representing our intention. It's also how we handle a letter suffix like 1.6.4a in other regexes (assuming `a` isn't short for alpha, `b` for beta, etc.).

This suggestion also uses the ternary approach to handling optional version parts, which is the typical pattern in similar casks (unless using the ternary format makes the line too long and hard to understand).